frontend/src/Pages/StudentProfile.tsx

```diff
@@ -0,0 +1,287 @@
+                                                        />
+                                                    </td>
+                                                </tr>
+                                            );
+                                        }
+                                    )}
+                                </tbody>
+                            </table>
+                        </div>
+                        {/* <div></div> */}
+                        <div className="card pt-2 px-3">
+                            <div className="text-teal-4 text-center text-lg font-semibold">
+                                Top 5 Recently Watched Videos
+                            </div>
+                            <table className="table-2 mb-4">
+                                <thead>
+                                    <tr className="grid-col-2">
+                                        <th className="justify-self-end">
+                                            Title
+                                        </th>
+                                        <th className="justify-self-start">
+                                            Rank
+                                        </th>
+                                    </tr>
+                                </thead>
+                                <tbody className="flex flex-col gap-4 mt-4">
+                                    {metrics.recent_videos.length > 0 ? (
+                                        metrics.recent_videos.map(
+                                            (
+                                                items: OpenContentResponse,
+                                                index: number
+                                            ) => {
+                                                return (
+                                                    <tr>
+                                                        <td className="justify-self-end">
+                                                            <img
+                                                                className="h-8 mx-auto object-contain"
+                                                                src={
+                                                                    items.thumbnail_url ??
+                                                                    ''
+                                                                }
+                                                            />
+                                                        </td>
+                                                        {items.title ??
+                                                            'Untitled'}
+                                                        <td className="justify-self-start">
+                                                            {index + 1}
+                                                        </td>
```

Comment:
We need to clamp the video title. My video is titled "2 Minutes Of Beautiful Scenery's On Planet Earth|Nature Is Power|Episode 01" and it doesn't fit. The text is allowed to be cut off and doesn't need to wrap.

Can we make the row clickable so that it navigates the user to that specific video?